packages/integrations/sitemap/src/index.ts

```diff
@@ -20,7 +20,7 @@ export type SitemapOptions =
 	| {
 			filenameBase?: string;
 			filter?(page: string): boolean;
-			customPages?: string[];
+			customPages?: string[] | ((page: string) => Promise<string[]> | string[]);
```

Comment:
This function can still return undefined, as you have described with the schema in Zod. I've been thinking about it for quite some time, and I think it might make sense for it to potentially return undefined in cases where the logic inside doesn’t actually do anything.

The curious part I haven’t been able to figure out yet is that if the promise returns undefined, the file isn’t generated, which makes me think it should definitely return something.

So you still need to handle that case properly, or make sure it always returns something.

```suggestion
			customPages?: string[] | ((page: string) => Promise<string[]> | string[] | undefined);
```